src/Core/Domain/Category/ValueObject/MenuThumbnailId.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Category\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Category\Exception\CategoryException;
+
+/**
+ * Stores id for category's menu thumbnail image.
+ */
+class MenuThumbnailId
+{
+    /**
+     * @var array category is of having maximum of 3 menu thumbnails with defined Ids
+     */
+    const ALLOWED_ID_VALUES = [0, 1, 2];
+
+    /**
+     * @var int
+     */
+    private $menuThumbnailId;
+
+    /**
+     * @param int $menuThumbnailId
+     */
+    public function __construct($menuThumbnailId)
+    {
+        $this->assertMenuThumbnailIsWithinAllowedValueRange($menuThumbnailId);
+
+        $this->menuThumbnailId = (int) $menuThumbnailId;
```

Comment:
same here, avoid casting

Reply:
problem is that data that comes from forms or query params are always string, the only solution i can think of is to cast id before passing to constructor, wdyt?

Reply:
👍